File: app/src/main/java/com/songwj/openvideo/mediacodec/encoder/Frame.java
package com.songwj.openvideo.mediacodec.encoder;

public class Frame {
    byte[] buffer;
    long presentationTimeUs;

    public byte[] getBuffer() {
        return buffer;
    }

    public void setBuffer(byte[] buffer) {
        this.buffer = buffer;
    }

    public long getPresentationTimeUs() {
        return presentationTimeUs;
    }

    public void setPresentationTimeUs(long presentationTimeUs) {
        this.presentationTimeUs = presentationTimeUs;
    }
}

File: app/src/main/java/com/songwj/openvideo/mediacodec/Frame.kt
package com.songwj.openvideo.mediacodec

import android.media.MediaCodec
import java.nio.ByteBuffer


/**
 * 一帧数据
 */
class Frame {
    var buffer: ByteBuffer? = null
    var bufferInfo = MediaCodec.BufferInfo()
        private set

    fun setBufferInfo(info: MediaCodec.BufferInfo) {
        bufferInfo.set(info.offset, info.size, info.presentationTimeUs, info.flags)
    }
}